### DISCUSSION TITLE
Added preview of doc, PDF, Excel, PPT attachments

### DISCUSSION THREAD

**Discussion by wojiaozhangyang** [outline/outline#3542] [Ideas]
I think the outline needs to add the preview of the attachment。
Direct previews for Doc, PPT, Excel, PDF, not Google Doc online documents  .
Whether this feature can be improved?

**Comment by almereyda** [comment]
Certainly PDF support could be possible through `pdf.js`. I don't know about the other formats.

**Comment by wojiaozhangyang** [comment]
react-file-viewer ： https://www.npmjs.com/package/react-file-viewer 
This library seems to solve a lot of problems

**Comment by HectorB-2020** [comment]
Preview would be great indeed! I would also want to be able to import Excel files. I can import Word (**.docx**) but not Excel (**.xlsx**).
Going further into OSS, I'd vote for importing LibreOffice Calc (**.ods**) and LibreOffice Writer (**.odt**).

**Comment by aariacarterweir** [comment]
Just wanna add to this, it'd be a neat feature!

**Comment by Luci2017** [comment]
Yes, really need this fuction !

**Comment by jnaubry** [comment]
Hi all,
I know it's an old topic but, still, this feature is not yet available on self hosted versions. Is there any chance to get it in a near future?
Thx

**Reply by tommoor** [reply]
It's probably unlikely to happen as these are all proprietary formats – I don't think there is a reasonable way to preview these in open source software.

**Reply by jnaubry** [reply]
ok, thanks for your quick reply. What about PDF format?

**Reply by tommoor** [reply]
Most browsers support previewing PDF so this might be doable, some way to expand PDF uploads to an inline version

**Comment by ballerbude** [comment]
Having a preview options for Excel and Word files would be excellent. LibreOffice/Onlyoffice can handle those files just fine, and those are FOSS tools. So I'm quite sure Outline could implement the preview feature for that files the FOSS way.

**Reply by T-Herrmann-WI** [reply]
I think Onlyoffice is the way to go in connection with the onlyoffice document server.
